### ISSUE TITLE
Allow Cardinal to run NekRS without any data transfers 

### ISSUE THREAD

**Issue by aprilnovak** [neams-th-coe/cardinal#161]
The `NekRSProblem` class is currently designed such that there are data transfers in (heat flux for CHT coupling, heat source for fission power coupling) and out (temperature) of NekRS for every time step. In addition, the problem setup performs many error checks on the NekRS input files, such as ensuring that you do have a `[TEMPERATURE]` block (so that you don't get seg faults when you try to index into `nrs->cds->S` as part of the data transfers.

There is interest, however, in using Cardinal exclusively for its data I/O and postprocessing features. I think it makes sense to add a `NekStandaloneProblem` class (which can have a common base class with `NekRSProblem` to share a lot of functionality) that:

- skips the entire `syncSolutions` function
- doesn't add `MooseVariables` for the coupling fields
- doesn't do any error checking of the NekRS input files in the context of a multiphysics problem
- doesn't pre-allocate `nrs->usrwrk`

Basically, this new `NekStandaloneProblem` should be able to run any NekRS input files out-of-the-box, even if those input files don't have a temperature solve. The goal would be to run a few of the NekRS example cases with a thin wrapper input so that those wanting to use Cardinal for I/O and postprocessing aren't limited by a number of multiphysics requirements we currently impose to catch user errors.

Having a common base class will also be helpful for when we eventually add SAM coupling, since functions such as `ExternalProblem::solve()` likely don't need to be implemented separately for 1-D loop coupling vs. 3-D coupling.
